tests/testSpanSets.py

```diff
@@ -0,0 +1,298 @@
+
+        self.assertEqual(quad.getIxx(), 0.4)
+        self.assertEqual(quad.getIyy(), 0.4)
+        self.assertEqual(quad.getIxy(), 0)
+
+    def testDilate(self):
+        spanSetPreDilate = afwGeom.SpanSet.spanSetFromShape(1, afwGeom.Stencil.CIRCLE)
+        spanSetPostDilate = spanSetPreDilate.dilate(1)
+
+        bBox = spanSetPostDilate.getBBox()
+        self.assertEqual(bBox.getMinX(), -2)
+        self.assertEqual(bBox.getMinY(), -2)
+
+    def testErode(self):
+        spanSetPreErode = afwGeom.SpanSet.spanSetFromShape(2, afwGeom.Stencil.CIRCLE)
+        spanSetPostErode = spanSetPreErode.erode(1)
+
+        bBox = spanSetPostErode.getBBox()
+        self.assertEqual(bBox.getMinX(), -1)
+        self.assertEqual(bBox.getMinY(), -1)
+
+    def testFlatten(self):
+        # Give an initial value to an input array
+        inputArray = np.ones((6, 6)) * 9
+        inputArray[1, 1] = 1
+        inputArray[1, 2] = 2
+        inputArray[2, 1] = 3
+        inputArray[2, 2] = 4
+
+        inputSpanSet = afwGeom.SpanSet([afwGeom.Span(0, 0, 1), afwGeom.Span(1, 0, 1)])
+        flatArr = inputSpanSet.flatten(inputArray, afwGeom.Point2I(-1, -1))
+
+        self.assertEqual(flatArr.size, inputSpanSet.getArea())
+
+    def testUnflatten(self):
+        inputArray = np.ones(6) * 4
+        inputSpanSet = afwGeom.SpanSet([afwGeom.Span(9, 2, 3),
+                                        afwGeom.Span(10, 3, 4),
+                                        afwGeom.Span(11, 2, 3)])
+        outputArray = inputSpanSet.unflatten(inputArray)
+
+        arrayShape = outputArray.shape
+        bBox = inputSpanSet.getBBox()
+        self.assertEqual(arrayShape[0], bBox.getHeight())
+        self.assertEqual(arrayShape[1], bBox.getWidth())
+
+    def populateMask(self):
+        msk = afwImage.MaskU(10, 10, 1)
+        spanSetMask = afwGeom.SpanSet.spanSetFromShape(3, afwGeom.Stencil.CIRCLE).shiftedBy(5, 5)
+        spanSetMask.setMask(msk, 2)
+        return msk, spanSetMask
+
+    def testSetMask(self):
+        mask, spanSetMask = self.populateMask()
+        mskArray = mask.getArray()
+        for i in range(mskArray.shape[0]):
+            for j in range(mskArray.shape[1]):
+                if afwGeom.Point2I(i, j) in spanSetMask:
+                    self.assertEqual(mskArray[i, j], 3)
+                else:
+                    self.assertEqual(mskArray[i, j], 1)
+
+    def testClearMask(self):
+        mask, spanSetMask = self.populateMask()
+        spanSetMask.clearMask(mask, 2)
+        mskArray = mask.getArray()
+        for i in range(mskArray.shape[0]):
+            for j in range(mskArray.shape[1]):
+                self.assertEqual(mskArray[i, j], 1)
+
+    def makeOverlapSpanSets(self):
+        firstSpanSet = afwGeom.SpanSet.spanSetFromShape(2, afwGeom.Stencil.BOX).shiftedBy(2, 4)
+        secondSpanSet = afwGeom.SpanSet.spanSetFromShape(2, afwGeom.Stencil.BOX).shiftedBy(2, 2)
+        return firstSpanSet, secondSpanSet
```

Comment:
For this (and the next) methods I would offset the SpanSets in the x direction, just to verify that intersection, union, etc are working properly.